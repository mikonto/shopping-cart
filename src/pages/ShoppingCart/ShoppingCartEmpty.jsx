import styled from "styled-components";
import { Link } from "react-router-dom";

const StyledShoppingCartEmpty = styled.div`
  background: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 12px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  border-radius: 4px;
`;

const BlueButton = styled.button`
  background-color: skyblue;
  padding: 8px 12px;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  transition: background-color 0.3s ease;

  &:hover {
    background-color: #6ca5bc;
  }
`;

const Spacer = styled.div`
  height: 15px;
`;

const Title = styled.h1`
  background: #f6f6f6;
  padding: 12px;
  margin: 8px;
`;

const ShoppingCartEmpty = () => {
  return (
    <StyledShoppingCartEmpty>
      <p>Your shopping cart is empty</p>
      <Link to="/products">
        <BlueButton>Shop now</BlueButton>
      </Link>
      <Spacer />
    </StyledShoppingCartEmpty>
  );
};

export default ShoppingCartEmpty;
