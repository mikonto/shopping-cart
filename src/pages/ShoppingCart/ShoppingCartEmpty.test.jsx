import { render, screen } from "@testing-library/react";
import ShoppingCartEmpty from "./ShoppingCartEmpty";
import { MemoryRouter } from "react-router-dom";

describe("ShoppingCartEmpty component", () => {
  it("renders the correct paragraph text", () => {
    render(
      <MemoryRouter>
        <ShoppingCartEmpty />
      </MemoryRouter>
    );
    const paragraphText = /your shopping cart is empty/i;
    const paragraph = screen.getByText(paragraphText);
    expect(paragraph).toBeInTheDocument();
  });

  it("renders the button and navigates correctly", () => {
    render(
      <MemoryRouter>
        <ShoppingCartEmpty />
      </MemoryRouter>
    );

    const button = screen.getByRole("button", { name: "Shop now" });
    expect(button).toBeInTheDocument();

    const link = screen.getByRole("link", { name: "Shop now" });
    expect(link).toHaveAttribute("href", "/products");
  });

  it("it renders correctly", () => {
    const { container } = render(
      <MemoryRouter>
        <ShoppingCartEmpty />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
