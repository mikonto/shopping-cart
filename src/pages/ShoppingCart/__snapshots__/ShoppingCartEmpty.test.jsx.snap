// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ShoppingCartEmpty component > it renders correctly 1`] = `
<div>
  <div
    class="sc-bcXGCL dCiGnF"
  >
    <p>
      Your shopping cart is empty
    </p>
    <a
      href="/products"
    >
      <button
        class="sc-gswONk bOaZaE"
      >
        Shop now
      </button>
    </a>
    <div
      class="sc-dkrGBB dMZkSX"
    />
  </div>
</div>
`;
